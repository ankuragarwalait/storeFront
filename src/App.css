/* Global Element Styles ------------------------------------------------------------------------------*/
body {
  margin: 0;
  padding: 0;
  font-family: sans-serif;
}

.App {
  text-align: center;
}

hr {
  margin: 1rem 0;
}

img.img-responsive {
  width: 100%;
  max-width: 100%;
}

.btn {
  display: block;
  padding: 1rem 3rem;
  font-size: 0.75rem; 
  color: white;
  border-width: 0;
  cursor: pointer;
  text-decoration: none;
}

.btn-primary {
  background-color: black;
}

.btn-secondary {
  background-color: grey;
}

.btn-tertiary {
  background-color: whote;
  color: black;
  border: 2px solid black;
}

.caret-link {
  cursor: pointer;
}

.caret-link::after {
  content: "";
  display: inline-block;
  margin-left: 0.2rem;
  vertical-align: middle;
  border-left: 0.2rem solid transparent;
  border-right: 0.2rem solid transparent;
  border-top: 0.2rem solid black;
}

/* Grid Styles ------------------------------------------------------------------------------ */

.container {
  max-width: 75rem;
  margin: 0 auto;
}

.row{
  display: flex;
  flex-wrap: wrap;
  padding: 0;
  justify-content: space-between;
}

.col-1 {
  flex: 1 0 8.33%;
}

.col-2 {
  flex: 1 0 16.66%;
}

.col-4 {
  flex: 1 0 30%;
  margin-right: 3.33%;
}

.col-6 {
  flex: 1 0 50%;
}

.col-8 {
  flex: 1 0 66.66%;
}

.col-4:nth-of-type(3n+3) {
  margin-right: 0;
}

/* Alignment Styles ------------------------------------------------------------------------------*/

.justify-center {
  justify-content: center;
}

.align-center {
  align-items: center;
}

.justify-around {
  justify-content: space-around;
}

.text-left {
  text-align: left;
}

.text-right {
  text-align: right;
}

/* Typography and Theme ------------------------------------------------------------------------------*/

.t-price, .t-brand {
  color: gray;
}

.t-brand {
  font-size: .75rem;
  font-weight: 500;
}

.t-font-xs {
  font-size: .25rem;
}

.t-font-s {
  font-size: .5rem;
}

.t-font-m {
  font-size: .75rem;
}

.t-cancel {
  float: right;
  cursor: pointer;
  color: gray;
  font-weight: bold;
}

.t-border-grey {
  border-color: grey;
}

.t-border-light-grey {
  border-color: lightgrey;
}

.t-color-black {
  color: black;
}

.t-color-white {
  color: white;
}

.t-color-grey {
  color: grey;
}

.t-color-lightgrey {
  color: lightgrey;
}
