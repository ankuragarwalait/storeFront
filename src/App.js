import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import './App.css';
import Category from "./category/Category";
import Cart from "./cart/Cart";
import Product from "./product/Product";
import Header from './shared/containers/Header';

class App extends Component {
  render() {
    return (
      <div className="App">    
        <Header cartCount={0} />
        <Switch>
          <Route exact path="/" component={Category} />
          <Route path="/cart" component={Cart}/>
          <Route path="/product/:id" component={Product}/>
        </Switch>
      </div>
    );
  }
}

export default App;
