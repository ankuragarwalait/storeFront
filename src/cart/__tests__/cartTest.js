import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import ConnectedCart, {Cart} from '../Cart';
import { BrowserRouter } from 'react-router-dom';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {mount} from 'enzyme';
import { mockStore } from '../../App.test';


let tree = null;
let store = null;

describe('test connected cart component', () => {
    beforeEach(()=> {
        configure({ adapter: new Adapter() });
        store = mockStore;
    });

    it('cart component rendered snapshot', ()=> {
        tree = renderer.create(<BrowserRouter>
            <Provider store={store}>
                <ConnectedCart />
            </Provider>
            </BrowserRouter>);
        const prodJson = tree.toJSON();
        expect(prodJson).toMatchSnapshot();
    });  

    it('test empty cart', ()=> {
        const props = {
            productList: {},
            total: 0
        }
        tree = mount(<BrowserRouter>
                        <Cart {...props} />
                    </BrowserRouter>);

        expect(tree.find("h2").props().children).toBe('No Products In Cart Yet.');
    })
});
