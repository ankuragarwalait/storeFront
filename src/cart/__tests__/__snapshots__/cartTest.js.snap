// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test connected cart component cart component rendered snapshot 1`] = `
<div
  className="Cart"
>
  <h1
    className="cart-heading"
  >
    Shopping Cart
  </h1>
  <div
    className="container cart-container"
  >
    <div
      className="row items-header t-color-lightgrey"
    >
      <div
        className="col-6 text-left"
      >
        PRODUCT
      </div>
      <div
        className="col-2"
      >
        QUANTITY
      </div>
      <div
        className="col-1"
      >
        TOTAL
      </div>
      <div
        className="col-1 text-right"
      >
        ACTION
      </div>
    </div>
    <hr />
    <div
      className="products-list-container"
    >
      <div
        className="row"
      >
        <div
          className="col-6 text-left"
        >
          <img
            alt="mashiko-yaki-indigo-small-plate"
            className="img-responsive"
            src="/media/mashiko-yaki-indigo-small-plate.jpg"
          />
          <div
            className="cart-prod-details"
          >
            <p
              className="t-brand"
            >
              Kiriko
            </p>
            <p>
              Mashiko-Yaki Indigo Small Plate
            </p>
          </div>
        </div>
        <div
          className="col-2"
        >
          <div>
            <input
              className="counter-field"
              readOnly={true}
              value={2}
            />
            <div
              className="counter-control"
            >
              <button
                onClick={[Function]}
              >
                +
              </button>
              <button
                className="btn-dec"
                onClick={[Function]}
              >
                -
              </button>
            </div>
          </div>
        </div>
        <div
          className="col-1 t-price"
        >
          $20.00
        </div>
        <div
          className="col-1 text-right"
        >
          <span
            className="t-cancel"
            onClick={[Function]}
          >
            ✕
          </span>
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="col-6 text-left"
        >
          <img
            alt="mashiko-yaki-saucer"
            className="img-responsive"
            src="/media/mashiko-yaki-saucer.jpg"
          />
          <div
            className="cart-prod-details"
          >
            <p
              className="t-brand"
            >
              Kiriko
            </p>
            <p>
              Mashiko-Yaki Saucer
            </p>
          </div>
        </div>
        <div
          className="col-2"
        >
          <div>
            <input
              className="counter-field"
              readOnly={true}
              value={1}
            />
            <div
              className="counter-control"
            >
              <button
                onClick={[Function]}
              >
                +
              </button>
              <button
                className="btn-dec"
                onClick={[Function]}
              >
                -
              </button>
            </div>
          </div>
        </div>
        <div
          className="col-1 t-price"
        >
          $10.00
        </div>
        <div
          className="col-1 text-right"
        >
          <span
            className="t-cancel"
            onClick={[Function]}
          >
            ✕
          </span>
        </div>
      </div>
    </div>
    <div
      className="row cart-overview"
    >
      <div
        className="col-6"
      >
         
      </div>
      <div
        className="col-4 text-left"
      >
        <p>
          CART OVERVIEW
        </p>
        <p>
          SUBTOTAL
        </p>
        <p>
          TOTAL
        </p>
      </div>
      <div
        className="col-2 text-right"
      >
        <p>
           
        </p>
        <p>
          $30.00
        </p>
        <p>
          $30.00
        </p>
      </div>
    </div>
    <hr />
    <div
      className="row"
    >
      <a
        href="/"
        onClick={[Function]}
      >
        Continue Shopping
      </a>
      <button
        className="btn btn-primary"
      >
        CHECKOUT 
        <span>
          (
          $30.00
          )
        </span>
      </button>
    </div>
  </div>
</div>
`;
